lib/react/rails/engine.rb

```diff
@@ -3,7 +3,12 @@ module Rails
     class Engine < ::Rails::Engine
       initializer "react_rails.setup_engine", :group => :all do |app|
         sprockets_env = app.assets || Sprockets # Sprockets 3.x expects this in a different place
-        sprockets_env.register_engine(".jsx", React::JSX::Template)
+        if Gem::Version.new(Sprockets::VERSION) >= Gem::Version.new("3.0.0")
+          sprockets_env.register_mime_type("application/jsx", extensions: [".jsx", ".js.jsx"])
```

Comment:
Thanks for the explanation! I see why you chose to do it this way and it sounds good to me!